Create and Swap Sass Color Variable in project.html - #4138

perlaroyerc opened the pull request:
Fixes #4138 

### What changes did you make and why did you make them?

  - Created "color-cyanblue" variable in variables/_colors.scss file.
  - Replaced the hardcoded color in _layouts/project.html with the created color variable.
  - Why? The Sass variable can be reused throughout the project resulting in a more maintainable and clean codebase.

No visual changes to the website.

perlaroyerc committed:
Updated current project team on website

perlaroyerc committed:
Merge origin into local branch

perlaroyerc committed:
Merge branch 'gh-pages' of https://github.com/hackforla/website into gh-pages

perlaroyerc committed:
Merge branch 'upstream-gh-pages' into gh-pages

perlaroyerc committed:
Created sass color vairable and added it to project.html file

Skydodle commented:
Availability: Monday 3/13/23 6 to 8pm
ETA: EOD Monday 3/13/23

roslynwythe commented:
Availability: 3/12 7-9 pm, 10 - noon 3/13
ETA EOD 3/13

roslynwythe commented on _layouts/project.html:
```diff
@@ -168,7 +168,7 @@ <h2 id='contributor-header' class='project-page-card-title title6'>All-Time Cont
       width="100%"
       height="100%"
-      style="margin: 0; border: 2px solid #336699; box-sizing: border-box">
+      style="margin: 0; border: 2px; color: color-cyanblue; box-sizing: border-box">
```
Hi @perlaroyerc great job on defining the new color and modifying the existing styles.  Currently the element you modified is not being rendered on the page, due to the liquid test, and so it is difficult to test in the browser.  This is an extremely minor point but I believe your statement would be cleaner and more consistent with the original syntax if you continued to use the 'border shorthand' format, like this:
`style="margin: 0; border: 2px solid color-cyanblue; box-sizing: border-box">`

roslynwythe replied:
Hi @perlaroyerc great job defining the new sass variable for the color, and modifying the html to use the variable.  

I have two minor requests to make.  First, please use a more specific name for the color variable, to distinguish it from other variations of cyan blue which we might use in the future.  The color is a medium dark cyan blue, so I suggest something like color-med-dark-cyanblue.  

Next, I believe it would be a cleaner solution to retain the border shorthand syntax used in the original code, and to explicitly specify a solid border, as follows:
`style="margin: 0; border: 2px solid color-med-dark-cyanblue; box-sizing: border-box">`

roslynwythe commented:
Hi @jdingeman this issue modified styling of iframes on the projects page, but it appears that due to the liquid condition `{% if page.iframe.available == true %}`, there are no instances of iframes appearing on the projects page.  If the iframe will no longer be used, we might want to create another issue to clean up the code by removing the liquid condition and html inside it.

Skydodle commented:
Hi @perlaroyerc, great job on creating a new color variable and replacing the hardcoded values.

Aside from what @roslynwythe mentioned about using the border shorthand syntax as well as retaining the solid styling for the border from the original code, I also recommend that you add the pull request to the project board. You can find instructions [here](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#v-complete-pull-request-5-how-to-add-a-pull-request-to-the-project-board) in the contirbuting.md if you need any guidance.

Thank you so much for your help!

perlaroyerc committed:
Merge branch 'upstream-gh-pages' into gh-pages

perlaroyerc committed:
Merge branch 'gh-pages' into design-systems-color-variable-4138

perlaroyerc committed:
Renamed sass variable and updated border shorthand syntax

perlaroyerc commented:
Hi! Thank you @roslynwythe and @Skydodle for your input. I have renamed the variable name to "mediumdarcyanblue" as it looked more cohesive with the rest of the color variables and updated the border syntax. I also added the PR to the Project Board, as Johnny pointed out.

Skydodle reviewed (approved):
I see that the color variable mediumdarkcyanblue was created for a more distinguishable variant of cyan blue,  border was changed to shorthand syntax correctly that includes the solid style, and PR was added to project board.

Great job! @perlaroyerc

roslynwythe reviewed (approved):
Thank you @perlaroyerc for a great job on this issue.  The branches are correct, the code is clean, and it looks great in the browser.
